packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -1191,6 +1206,39 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         event.preventDefault();
     }
 
+    /**
+     * Handle the single-click mouse event on the expansion toggle.
+     * @param node the tree node if available.
+     * @param event the mouse single-click event.
+     */
+    protected handleExpansionToggleClickEvent(node: TreeNode | undefined, event: React.MouseEvent<HTMLElement>): void {
+        if (node) {
+            if (!!this.props.multiSelect) {
+                const shiftMask = this.hasShiftMask(event);
+                const ctrlCmdMask = this.hasCtrlCmdMask(event);
+                if (this.isExpandable(node) && !shiftMask && !ctrlCmdMask) {
+                    this.model.toggleNodeExpansion(node);
+                }
+            } else {
+                if (this.isExpandable(node) && !this.hasCtrlCmdMask(event) && !this.hasShiftMask(event)) {
+                    this.model.toggleNodeExpansion(node);
+                }
+            }
+            event.stopPropagation();
+        }
+    }
+
+    /**
+     * Handle the double-click mouse event on the expansion toggle.
+     * @param event the double-click mouse event.
+     */
+    protected handleExpansionToggleDblClickEvent(event: React.MouseEvent<HTMLElement>): void {
```

Comment:
This may present an opportunity to improve the default behavior of the tree, as well. At the moment, double-clicking on a folder node in the explorer triple toggles the node (once for each of the two single clicks and once for the double click) where I think any reasonable user would expect a double-toggle. There have been an number of attempts to address aspects of that behavior (most recently by yours truly #9567). Is there any way to use this new method to modify that behavior?